### ISSUE TITLE
Support for workspace symbols

### ISSUE THREAD

**Issue by jgarte** [python-lsp/python-lsp-server#237]
I ask because I tried using [this package](https://github.com/mohkale/consult-eglot) and it complained that `pylsp` does not support this feature. It worked well with pyright though.

Does pyright support symbols and pylsp not?

**Comment by ccordoba12** [comment]
Correct, Pylsp doesn't support workspace symbols, sorry.

**Comment by jgarte** [comment]
Is there any plans to implement it?

@ccordoba12 

What's you approach to workspace symbols? Do you prefer not to use this feature?

**Comment by ccordoba12** [comment]
There are no plans right now, sorry.

Maybe Jedi offers this functionality but we haven't checked because there has been little interest for it.

**Comment by astoff** [comment]
For the record, I find the workspace symbols feature quite useful. One can always grep the project for `def something`, but that's not as handy. It would also seem relatively straightforward to implement, which gives it a pretty good cost/benefit ratio.

**Comment by ccordoba12** [comment]
It seems this is supported by Jedi, so it's feasible here. However, we're swamped with other things, so I don't know when we'll have time to implement it.

**Comment by tristone13th** [comment]
> 

In the README I saw you mentioned that Jedi can provide this function, so does it mean we can use it well now?

**Comment by ccordoba12** [comment]
No, we need to implement that functionality in the way expected by the LSP protocol here.

**Comment by jgarte** [comment]
> No, we need to implement that functionality in the way expected by the LSP protocol here.

@ccordoba12 Thanks for clarifying that for us.

**Comment by eyalk11** [comment]
Made a pull request to resolve it. https://github.com/python-lsp/python-lsp-server/pull/511

**Comment by niilohlin** [comment]
https://github.com/python-lsp/python-lsp-server/pull/564

I'm trying to make an effort to solve this too. Although I have some issues with just setting up pluggy and calling the hooks. If anyone has time to take a look and point me in the right direction that would be much appreciated.

**Comment by Cadair** [comment]
Just wanted to politely bump this, I would find it very useful and would love to see an implementation merged.

**Comment by agzam** [comment]
Seriously? How the heck an lsp-server doesn't support one of the most important features? Kind of a break-dealer, no?

**Comment by savchenko** [comment]
OK, which Python LSP server provides workspace symbols?

**Comment by agzam** [comment]
> OK, which Python LSP server provides workspace symbols?

basedpyright does. 

_upd: I initially replied from my phone, came to find my laptop to edit my own comment for clarification._ Even though [my initial question](https://github.com/python-lsp/python-lsp-server/issues/237#issuecomment-2776913934) may look like demanding and kind of a dick comment, it was genuine. I don't even write Python day-to-day, I just needed to explore a couple of projects done by another team and I wished for lsp support. I found this project and the number of interesting features implemented in it got me excited. I wasn't even disappointed, but rather confused and surprised by the lack of this feature. Project-wide workspace symbol lookup is one of the most used features for me in other languages I use, so my confusion I believe was justified. 

I posted my question because I thought maybe, there's an easy workaround. Turns out basedpyright supports the feature, but if it gets implemented here, I would happily use it. Like I said, I have no horses in this race, I'm not actively writing Python - I don't really care for the best and most complete implementation of all the lsp features in a server, but particularly this feature, for me personally is somewhat important.

**Comment by jbirnick** [comment]
Would love to have workspace symbols too, I use them all the time in my development workflow!
